### PR TITLE
Sparkbox Sprint 62

### PR DESCRIPTION
**Pull request by lunarfusion**
# READY FOR REVIEW

## Summary
* #1089
* #1087 
* #1097 
* #1095 
* #1098 
* #1093

TASK LIST:
- [x]  Add compiled CSS & JS
- [x]  Run BackstopJS visual regression tests
- [x]  Ensure Code Climate passes against the develop branch. https://codeclimate.com/github/SU-HSDO/suhumsci/pull/1086
- [x]  Assign to Mike for review

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. `npm run test` should pass in the theme directory
2. If you are interested in the testing steps for associated PRs, please take a look at the individual PR's description and "Steps to Test" section.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

File: docroot/themes/humsci/humsci_colorful/humsci_colorful.theme [hunk 1]
```diff
@@ -23,6 +23,13 @@ function humsci_colorful_preprocess_html(&$vars) {
     // Set html_attributes for html DOM element
     $vars['html_attributes']->addClass('hb-dark-inversion');
   }
+
+  // Mega Menu hides standard menu on desktop setting.
+  $use_megamenu = theme_get_setting('megamenu_toggle');
+  if ($use_megamenu) {
+    // Load base megamenu CSS when megamenu is enabled
+    $vars['#attached']['library'][] =  'humsci_colorful/megamenu';
```
Comment (pookmish) [inline docroot/themes/humsci/humsci_colorful/humsci_colorful.theme]: This should be added at the block layer using a `preprocess_block__we_megamenu` I believe. Then you don't need to have the condition if the theme has it turned on

File: docroot/themes/humsci/humsci_traditional/humsci_traditional.theme [hunk 2]
```diff
@@ -30,6 +30,13 @@ function humsci_traditional_preprocess_html(&$vars) {
     // Set html_attributes for html DOM element
     $vars['html_attributes']->addClass('hb-dark-inversion');
   }
+
+  // Mega Menu hides standard menu on desktop setting.
+  $use_megamenu = theme_get_setting('megamenu_toggle');
+  if ($use_megamenu) {
+    // Load base megamenu CSS when megamenu is enabled
+    $vars['#attached']['library'][] =  'humsci_traditional/megamenu';
```
Comment (pookmish) [inline docroot/themes/humsci/humsci_traditional/humsci_traditional.theme]: This should be added at the block layer using a `preprocess_block__we_megamenu` I believe. Then you don't need to have the condition if the theme has it turned on

**Review by pookmish — CHANGES_REQUESTED** [review]

File: docroot/themes/humsci/humsci_colorful/humsci_colorful.theme (hunk 1 above)
Comment (pookmish) [reply docroot/themes/humsci/humsci_colorful/humsci_colorful.theme]: alternatively, you could alter the library using a `hook_library_info_alter` and add the dependency

File: docroot/themes/humsci/humsci_traditional/humsci_traditional.theme (hunk 2 above)
Comment (lunarfusion) [reply docroot/themes/humsci/humsci_traditional/humsci_traditional.theme]: Please take a look at the latest commit. I've changed it to we_megamenu_preprocess_block and tested Traditional and Colorful with the mega menu disabled and enabled. Appears to load the base CSS library only when it's enabled as hoped.

Comment (pookmish) [reply docroot/themes/humsci/humsci_traditional/humsci_traditional.theme]: Now this function will never run. A hook does not begin like this. Target the specific megemu block and rove the theme get settings line

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1086-8devbcicaprhmed2qudbjeo9lzsgsxeh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6256f3b07a1d628954c46fe7

**Review by pookmish — APPROVED** [review]
Thank you. This is the correct way.

**Pull request merged**